homeassistant/components/backup/manager.py

```diff
@@ -1544,6 +1560,47 @@ def _write_restore_file() -> None:
         )
         await self._hass.services.async_call("homeassistant", "restart", blocking=True)
 
+    async def async_resume_restore_progress_after_restart(
+        self,
+        *,
+        on_progress: Callable[[RestoreBackupEvent | IdleEvent], None],
+    ) -> None:
+        """Check restore status after core restart."""
+
+        def _read_restore_file() -> json_util.JsonObjectType | None:
+            """Read the restore file."""
+            result_path = Path(self._hass.config.path(RESTORE_BACKUP_RESULT_FILE))
+
+            try:
+                restore_result = json_util.json_loads_object(result_path.read_bytes())
+            except FileNotFoundError:
+                return None
+            finally:
+                result_path.unlink(missing_ok=True)
```

Comment:
Do we need to catch OSError when removing the file?

Reply:
Yeah

Reply:
I added a commit which catches `OSError` and logs a warning